app/src/main/java/org/fossasia/openevent/app/data/EventDataRepository.java

```diff
@@ -83,6 +83,18 @@ public EventDataRepository(IUtilModel utilModel) {
         return getData(() -> eventService.getEvents(authorization), EVENTS, reload);
     }
 
+    @Override
+    public Observable<Attendee> toggleAttendeeCheckStatus(long position, long eventId, long attendeeId) {
+        return eventService.toggleAttendeeCheckStatus(eventId, attendeeId, authorization)
+            .doOnNext(attendee -> {
+                List<Attendee> attendees = (List<Attendee>) cacheModel.getValue(ATTENDEES + eventId);
```

Comment:
You can't rely, that on return of this function, cache will give you a non null attendee list.

What if it was never saved or cleared due to some reason like force reload?